diff --git a/lua/galaxyline/provider.lua b/lua/galaxyline/provider.lua
index 8d497f0..dfa77c8 100644
--- a/lua/galaxyline/provider.lua
+++ b/lua/galaxyline/provider.lua
@@ -11,7 +11,7 @@ local function load_providers()
     BufferIcon = buffer.get_buffer_type_icon,
     BufferNumber = buffer.get_buffer_number,
     FileTypeName = buffer.get_buffer_filetype,
-    GitBranch = vcs.get_git_branch,
+    GitBranch = vcs.get_git_short_branch,
     DiffAdd = vcs.diff_add,
     DiffModified = vcs.diff_modified,
     DiffRemove = vcs.diff_remove,
diff --git a/lua/galaxyline/providers/vcs.lua b/lua/galaxyline/providers/vcs.lua
index 63ca4c4..69ee259 100644
--- a/lua/galaxyline/providers/vcs.lua
+++ b/lua/galaxyline/providers/vcs.lua
@@ -134,6 +134,15 @@ local function get_git_head_state(git_dir)
   end
 end
 
+vcs.get_git_short_branch = function()
+    local branch = vcs.get_git_branch()
+    if string.len(branch) > 20 then
+        branch = string.sub(branch,1,17) .. '...'
+    end
+
+    return branch
+end
+
 vcs.get_git_branch = function()
   -- Try to get branch with gitsigns first before trying to get it directly with Git
   if vim.b.gitsigns_head and string.len(vim.b.gitsigns_head) > 0 then
diff --git a/lua/galaxyline/themes/eviline.lua b/lua/galaxyline/themes/eviline.lua
index 0632962..621fecb 100644
--- a/lua/galaxyline/themes/eviline.lua
+++ b/lua/galaxyline/themes/eviline.lua
@@ -2,7 +2,7 @@ local gl = require("galaxyline")
 local colors = require("galaxyline.themes.colors")["doom-one"]
 local condition = require("galaxyline.condition")
 local gls = gl.section
-gl.short_line_list = { "NvimTree", "vista", "dbui", "packer" }
+gl.short_line_list = { "NvimTree", "vista", "dbui", "packer", "DiffviewFileHistory", "DiffviewFiles"}
 
 gls.left[1] = {
   RainbowRed = {
